presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -410,6 +407,333 @@ Drop a schema::
+  this limit, writers will stop writing new splits until some of hteme are used by workers.
+  Higher values will increase memory usage, but allow IO to be concentrated at one time,
+  which may be faster and increase resource utilization.
+
+
+``hive.max-partitions-per-writers``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:**
+
+  Maximum number of partitions per writer. A query will fail if it requires more partitions
+  per writer than allowed by this property. It can be helpful to have queries beyond the
+  expected maximum partitions to fail to help with error detection. Also it may allow to
+  preactivly avoid out of memory problem.
+
+
+``hive.max-split-iterator-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:**
+
+  This property describes how many threads may be used to iterate through splits when loading
+  them to the worker nodes. A higher value may increase parallelism, but increased concurrency
+  may cause too much time to be spent on context switching.
+
+
+``hive.max-split-size``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``64 MB``
+ * **Description:**
+
+  The maximum size of splits created after the initial splits. The logic for initial splits is
+  described in ``hive.max-initial-splits``. A higher value will reduce parallelism. This may be
+  desirable for very large queries and a stable cluster because it allows for more efficient
+  processing of local data without the context switching, synchronization and data collection
+  that result from parallelization. The optimal value should be aligned with the average query
+  size in the system.
+
+
+``hive.metastore.partition-batch-size.max``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:**
+
+  This together with ``hive.metastore.partition-batch-size.min`` defines the range of partition
+  sizes read from Hive. The first partition is always of size ``hive.metastore.partition-batch-size.min``
+  and each following partition is two times bigger than previous up to
+  ``hive.mestastore.partition-batch-size.max`` (the formula for partition size ``n`` is
+  min(``hive.metastore.partition-batch-size.max``, (``2``^``n``) * ``hive.metastore.partition-batch-size.min``)).
+  This algorithm allows for live adjustment of partition size according to the processing requirements.
+  If the queries in the system will differ significantly from each other in size, then this range should be
+  extended to better adjust to processing requirements. If the queries in the system will mostly be of the
+  same size, then setting both values to the same maximally tuned value may give a slight edge in
+  processing time.
```

Comment:
I think it is also about responsiveness, if data to be read is small then it will be faster to load more smaller splits and execute the concurrently, but if query is processing big queries then it is better to have bigger splits (non iteractive query case).

Reply:
ya but thats obvious right? Do we want to add it to the description?

Reply:
ah... ok. Lets leave it as it is then.